packages/playground/src/weblets/tf_contracts_list.vue

```diff
@@ -465,12 +469,47 @@ const nodeStatus = computed(() => {
 // Calculate the total cost of contracts
 async function getTotalCost() {
   totalCost.value = 0;
-  for (const contract of contracts.value) {
-    totalCost.value = +new Decimal(totalCost.value).add(contract.consumption?.valueOf() || 0);
+
+  try {
+    const nodeResponse = await gridProxyClient.contracts.list({
+      twinId: profileManager.profile!.twinId,
+      state: [ContractState.Created, ContractState.GracePeriod],
+      type: ContractType.Node,
+      retCount: true,
+    });
+
+    const nameResponse = await gridProxyClient.contracts.list({
+      twinId: profileManager.profile!.twinId,
+      state: [ContractState.Created, ContractState.GracePeriod],
+      type: ContractType.Name,
+      retCount: true,
+    });
+
+    const rentResponse = await gridProxyClient.contracts.list({
+      twinId: profileManager.profile!.twinId,
+      state: [ContractState.Created, ContractState.GracePeriod],
+      type: ContractType.Rent,
+      retCount: true,
+    });
+
+    const allNodeContracts = await _normalizeContracts(nodeResponse.data, ContractType.Node);
+    const allNameContracts = await _normalizeContracts(nameResponse.data, ContractType.Name);
+    const allRentContracts = await _normalizeContracts(rentResponse.data, ContractType.Rent);
+
+    const allContracts = [...allNodeContracts, ...allNameContracts, ...allRentContracts];
+    for (const contract of allContracts) {
+      const consumption =
+        contract.consumption !== undefined && contract.consumption !== null ? contract.consumption.valueOf() : 0;
+      totalCost.value = +new Decimal(totalCost.value).add(consumption);
+    }
```

Comment:
The same logic of listing contracts is already used in the  grid client 
https://github.com/threefoldtech/tfgrid-sdk-ts/blob/b65f6b43b809181aebc274c5576b4c06726b9ead/packages/grid_client/src/modules/contracts.ts#L300-L312

Reply:
code redundance, the old function is already tested, and we are using it in many places so why do we need to introduce a new method that does the same exact thing?